app/services/process-payload.js

```diff
@@ -0,0 +1,44 @@
+const Project = require('../models/project')
```

Comment:
You could refactor and do something more funky by using `map` and `flatMap`.

```
const processProjects = (projects) => projects.map(project => {
  const projectData = new Project(project)
  projectData.sessions = processSessions(project.sessions)
  return projectData
})
```

And do the same down the chain for Sessions `processSessions`, Threads  `processThreads ` and Steps `processTreads`

Then you can use a `flatMap` to get each section `sessions`, 'threads` and `steps` at the end and load it into the repo actions.

```
const sessions = projects.flatMap(project => project.sessions)
```

```
await addSessions(sessions)
```